.github/workflows/test-python-app.yaml

```diff
@@ -0,0 +1,61 @@
+name: Test Python (Django) application
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test_django:
+
+    runs-on: ubuntu-latest
+    
+    # setup the postgres service needed by the webapp
+    services:
+      postgres:
+        image: postgres:12
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: mp
+        ports:
+          - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+        
+    # setup env for the application
+    env:
+      DJANGO_SETTINGS_MODULE: 'settings'
+      DB_HOST: 127.0.0.1
+      DB_PASSWORD: postgres
+      DB_NAME: mp
+      DB_USER: postgres
+      SECRET_KEY: thISkEyisUsedForTravis
+      SOLR_URL: http://127.0.0.1:8983/solr/tester
+      SOLR_ADMIN_URL: http://127.0.0.1:8983/solr/admin/cores
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```

Comment:
Not really important ...
Is such an old version used because that's what's available by default on Ubuntu 18.04?
(We could (or should) probably switch to 20.04 LTS anyway.)

Reply:
> I see that the job actually had troubles, but the pipeline passed?
> 
> ```
> ../opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/rest_framework/pagination.py:200: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'buildhistory.models.BuildHistory'> QuerySet.
>   paginator = self.django_paginator_class(queryset, page_size)
> ........../opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/rest_framework/pagination.py:200: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'category.models.Category'> QuerySet.
>   paginator = self.django_paginator_class(queryset, page_size)
> ....Failed to query Solr using 'active:(true)': Failed to connect to server at http://127.0.0.1:8983/solr/tester/select/?q=active%3A%28true%29&fl=%2A+score&df=text&sort=name_lower+asc&start=0&rows=20&facet=on&facet.field=maintainers_exact&facet.field=categories_exact&facet.field=variants_exact&fq=django_ct%3A%28port.port%29&fq=categories_exact%3A%22categoryA%22&wt=json: HTTPConnectionPool(host='127.0.0.1', port=8983): Max retries exceeded with url: /solr/tester/select/?q=active%3A%28true%29&fl=%2A+score&df=text&sort=name_lower+asc&start=0&rows=20&facet=on&facet.field=maintainers_exact&facet.field=categories_exact&facet.field=variants_exact&fq=django_ct%3A%28port.port%29&fq=categories_exact%3A%22categoryA%22&wt=json (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f6da1dc6110>: Failed to establish a new connection: [Errno 111] Connection refused'))
> Traceback (most recent call last):
>   File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/urllib3/connection.py", line 160, in _new_conn
>     (self._dns_host, self.port), self.timeout, **extra_kw
>   File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/urllib3/util/connection.py", line 84, in create_connection
>     raise err
>   File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/urllib3/util/connection.py", line 74, in create_connection
>     sock.connect(sa)
> ConnectionRefusedError: [Errno 111] Connection refused
> ```

This is not a problem with code or tests. It is expected behaviour as the app tries to connect to Solr, which does not exist for tests, and hence the warnings.

Reply:
> Not really important ...
> Is such an old version used because that's what's available by default on Ubuntu 18.04?
> (We could (or should) probably switch to 20.04 LTS anyway.)

I tried running the tests for 3.9 earlier, encountered 3-4 issues with `encoding` which would need to be fixed first. Also this Python version transition should also happen in the Dockerfile, some fixes are due for that.

Reply:
OK, thanks for the info. Yes, it makes sense to update things, that doesn't need to hold us back from migrating to the new app, in particular if updating doesn't work correctly. (None of my other comments are "serious" enough to hold us back.)